File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        
        unordered_map<char, int> sm, tm;
        
        for(int i = 0; i < s.length(); i++){
            if(sm.find(s[i]) == sm.end()) sm.insert({s[i], 1});
            else sm[s[i]]++;
            
            if(tm.find(t[i]) == tm.end()) tm.insert({t[i], 1});
            else tm[t[i]]++;
        }
        
        return (sm == tm);
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s) != len(t)):
            return False
        
        sm, tm = {}, {}
        for i in range(len(s)):
            sm[s[i]] = 1 + sm.get(s[i], 0)
            tm[t[i]] = 1 + tm.get(t[i], 0)
            
        return (sm == tm)